mkosi/__init__.py

```diff
@@ -2133,7 +2134,10 @@ def install_uki(
     with umask(~0o700):
         boot_binary.parent.mkdir(parents=True, exist_ok=True)
 
-    if context.config.bootloader.is_signed():
+    if (
+        context.config.bootloader.is_signed()
+        and context.config.unified_kernel_images != UnifiedKernelImage.unsigned
+    ):
```

Comment:
We should never go into the else branch here if `unified_kernel_images` is set to `signed`. 

So in this function I'd determine the final `UnifiedKernelImage` value by translating `auto` to `signed` if the bootloader is signed and `unsigned` if the bootloader is unsigned.

If not set to `auto` then we always use the configured value, so we'll install unsigned UKI with signed bootloader or signed UKI with unsigned bootloader. Might not make sense but if the user explicitly wants it we should do it.